lib/matplotlib/colorbar.py

```diff
@@ -294,6 +228,70 @@ def draw(self, renderer):
         return ret
 
 
+class _ColorbarAxesLocator:
+    """
+    Wrap any locator on the axes, or its position, to shrink the
+    inner axes if there are extend triangles at either min or max.
+    """
+    def __init__(self, cbar):
+        """
+        *bounds* (a ``[l, b, w, h]`` rectangle) and *transform* together
+        specify the position of the inset axes.
+        """
+        self._cbar = cbar
+        self._orig_locator = cbar.ax._axes_locator
+
+    def __call__(self, ax, renderer):
+
+        # make sure that lims and scales are the same
+        scale = self._cbar._long_axis().get_scale()
+        try:
+            self._cbar._short_axis()._set_scale(scale)
+        except TypeError:
+            pass
+        lim = self._cbar._long_axis().get_view_interval()
+        self._cbar._short_axis().set_view_interval(*lim)
+
+        if self._orig_locator is not None:
+            pos = self._orig_locator(ax, renderer)
+        else:
+            pos = ax.get_position(original=True)
+        if self._cbar.extend == 'neither':
+            return pos
+
+        y, extendlen = self._cbar._proportional_y()
+        if not self._cbar._extend_lower():
+            extendlen[0] = 0
+        if not self._cbar._extend_upper():
+            extendlen[1] = 0
+        len = sum(extendlen) + 1
+        shrink = 1 / len
+        offset = extendlen[0] / len
+        # we need to reset the aspect ratio of the axes to account
+        # of the extends...
+        if not self._cbar._userax:
```

Comment:
Do we need the `_userax` attribute anymore (it was added in your previous PR I believe)? I'm wondering if you can take advantage of `not hasattr("_colorbar_info")` rather than passing the keyword `userax` around.

Reply:
Good point.  Fixed.